public/components/custom_panels/helpers/utils.tsx

```diff
@@ -384,6 +391,183 @@ export const renderCatalogVisualization = async ({
+  const pattern = /\[Otel Metric\]\s(\S+?-\S+?)\.(\S+)/;
+  const match = metricString.match(pattern);
+
+  if (match) {
+    const index = match[1];
+    const documentName = match[2];
+    return [index, documentName];
+  } else {
+    return null;
+  }
+};
+
+export const renderOpenTelemetryVisualization = async ({
+  startTime,
+  endTime,
+  setVisualizationTitle,
+  setVisualizationType,
+  setVisualizationData,
+  setVisualizationMetaData,
+  setIsLoading,
+  setIsError,
+  visualization,
+}: {
+  startTime: string;
+  endTime: string;
+  setVisualizationTitle: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationType: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationData: React.Dispatch<React.SetStateAction<Plotly.Data[]>>;
+  setVisualizationMetaData: React.Dispatch<React.SetStateAction<undefined>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>;
+  visualization: any;
+}) => {
+  setIsLoading(true);
+  setIsError({} as VizContainerError);
+
+  const { http } = coreRefs;
+  const visualizationType = 'bar';
+  let index = visualization?.index;
+  let documentName = visualization?.name;
+
+  if (index === undefined) {
+    const indexAndDocumentName = extractIndexAndDocumentName(visualization.name);
+    index = indexAndDocumentName[0];
+    documentName = indexAndDocumentName[1];
+  }
+  const fetchSampleDocument = await fetchSampleOTDocument(index, http, documentName)();
+  const source = fetchSampleDocument.hits[0]._source;
+
+  setVisualizationType(visualizationType);
+  setVisualizationTitle(source.name);
+
+  const dataBinsPromises = source.buckets.map(async (bucket: any) => {
+    try {
+      const formattedStartTime = convertDateTime(startTime, false, false, false, true);
+      const formattedEndTime = convertDateTime(endTime, false, false, false, true);
+      const fetchingAggregatedBinCount = await fetchAggregatedBinCount(
+        bucket.min.toString(),
+        bucket.max.toString(),
+        formattedStartTime,
+        formattedEndTime,
+        documentName,
+        http,
+        index,
+        setIsError,
+        setIsLoading
+      )();
+
+      return {
+        xAxis: bucket.min + ' - ' + bucket.max,
+        'count()': fetchingAggregatedBinCount?.nested_buckets?.bucket_range?.bucket_count?.value,
+      };
+    } catch (error) {
+      console.error('Error processing bucket:', error);
+      return null;
+    }
+  });
+  const jsonData = await Promise.all(dataBinsPromises);
+  const formatDataBinsName = () => {
+    return { dataBins: { jsonData } };
+  };
+  const formatedJsonData = formatDataBinsName().dataBins;
+  const visualizationMetaData = createOtelVisualizationMetaData(
+    documentName,
+    visualizationType,
+    startTime,
+    endTime,
+    formatedJsonData
+  );
+  setVisualizationData(formatedJsonData);
+  setVisualizationMetaData(visualizationMetaData);
+};
+
```

Comment:
These all functions should not be part of `panels/utils.tsx`. These should be moved to `metrics/utils.tsx`